actionpack/test/dispatch/content_disposition_test.rb

```diff
@@ -28,6 +28,14 @@ class ContentDispositionTest < ActiveSupport::TestCase
       assert_equal "inline; #{disposition.ascii_filename}; #{disposition.utf8_filename}", disposition.to_s
     end
 
+    test "encoding a filename with permitted chars" do
+      disposition = Http::ContentDisposition.new(disposition: :inline, filename: "argh+!#$-123_|&^`~.jpg")
```

Comment:
This is not strictly necessary, since `#$-` doesn't seem to have the same interpolation problem as other patterns. But it may we worth defensively escaping it anyway in case anybody ever adds another character right after the dollar sign that does cause interpolation (e.g. `#$@`):

```suggestion
      disposition = Http::ContentDisposition.new(disposition: :inline, filename: "argh+!\#$-123_|&^`~.jpg")
```

Reply:
I tried to make this change, but rubocop complained. I can't see why (see attached photo).

![image](https://user-images.githubusercontent.com/122287/91724717-01749880-eb9e-11ea-9a6e-6c66cc576b6e.png)

Reply:
It doesn't like the single quotes:

```
 actionpack/test/dispatch/content_disposition_test.rb:32:82: C: Style/StringLiterals: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
      disposition = Http::ContentDisposition.new(disposition: :inline, filename: 'argh+!#$-123_|&^`~.jpg')
                                                                                 ^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
I've added a commit, rubocop job pending…